src/internal/operators/groupBy.ts

```diff
@@ -19,53 +19,70 @@ export function groupBy<T, K, R>(keySelector: (value: T) => K, elementSelector?:
  *
  * ![](groupBy.png)
  *
- * ##Examples
- * Group objects by id and return as array
- * ```typescript
+ * When the Observable emits an item, a key is computed for this item with the keySelector function.
+ *
+ * If a {@link GroupedObservable} for this key exists, this {@link GroupedObservable} emits. Elsewhere, a new
+ * {@link GroupedObservable} for this key is created and emits.
+ *
+ * A {@link GroupedObservable} represents values belonging to the same group represented by a common key. The common
+ * key is available as the key field of a {@link GroupedObservable} instance.
+ *
+ * The elements emitted by {@link GroupedObservable}s are by default the items emitted by the Observable, or elements
+ * returned by the elementSelector function.
+ *
+ * ## Examples
+ * ### Group objects by id and return as array
+ * ```javascript
+ * import { mergeMap, groupBy } from 'rxjs/operators';
+ * import { of } from 'rxjs/observable/of';
+ *
  * interface Obj {
  *    id: number,
  *    name: string,
  * }
  *
  * of<Obj>(
- *   {id: 1, name: 'aze1'},
- *   {id: 2, name: 'sf2'},
- *   {id: 2, name: 'dg2'},
- *   {id: 1, name: 'erg1'},
- *   {id: 1, name: 'df1'},
- *   {id: 2, name: 'sfqfb2'},
- *   {id: 3, name: 'qfs3'},
- *   {id: 2, name: 'qsgqsfg2'},
+ *   {id: 1, name: 'aze1' 'javascript'},
+ *   {id: 2, name: 'sf2' 'parcel'},
+ *   {id: 2, name: 'dg2' 'webpack'},
+ *   {id: 1, name: 'erg1' 'typescript'},
+ *   {id: 1, name: 'df1' 'ecmascript'},
+ *   {id: 2, name: 'sfqfb2' 'rollup'},
+ *   {id: 3, name: 'qfs3' 'tslint'},
+ *   {id: 2, name: 'qsgqsfg2' 'babel'},
  * ).pipe(
  *   groupBy(p => p.id),
  *   mergeMap((group$) => group$.pipe(reduce((acc, cur) => [...acc, cur], []))),
  * )
  * .subscribe(p => console.log(p));
  *
  * // displays:
- * // [ { id: 1, name: 'aze1' },
- * //   { id: 1, name: 'erg1' },
- * //   { id: 1, name: 'df1' } ]
+ * // [ { id: 1, name: 'javascript'},
```

Comment:
We try to have short api examples, so that you get the way this operator works as fast as possible. Maybe you can shorten it a bit

Reply:
@JWO719 I have shorted and fixed the first example as well.